### PR TITLE
Generalize the auth uri and fix trailing slash

### PR DESCRIPTION
**Pull request by PatrickEifler**

### CODE REVIEW

File: authorization/oauth2.js [hunk 1]
```diff
@@ -35,7 +35,7 @@ class Oauth2 {
             scope: 'access-own-user-info+configure-devices'
         };
 
-        let uri = `${this.uri}oauth2/auth?client_id=${this.appId}&redirect_uri=${this.redirectURI}&response_type=token&scope=access-own-user-info+configure-devices`;
+        let uri = `https://${this.uri}/oauth2/auth?client_id=${this.appId}&redirect_uri=${this.redirectURI}&response_type=token&scope=access-own-user-info+configure-devices`;
```
Comment (SmbatYeranyan) [inline authorization/oauth2.js]: This wont work if we are going to be using the auth class for multiple services. For example, on our batch-service dashboard im using https://api.relayr.io (HTTPS) and dev1-batch-api.relayr.io (not HTTPS) at the same time.

Comment (PatrickEifler) [inline authorization/oauth2.js]: So the protocol needs to be configureable?

Comment (PatrickEifler) [inline authorization/oauth2.js]: Now it can be passed in the customConfig as {protocol: 'something'} - It defaults to https

**Comment by SmbatYeranyan** [comment]
Nice! that should do it. Ready to merge

**Pull request merged**